### ISSUE TITLE
Package

### ISSUE THREAD

**Issue by marwie** [Unity-Technologies/NavMeshComponents#172]
Hello, would it be possible to provide a package with these components?

**Comment by PierreMizzi** [comment]
Yes please, something lighter

**Comment by junpeikun** [comment]
Or even ship it as an built-in package? Still surprised I did not find NavMeshSurface in the Add component dropdown 😐

**Comment by pedro15** [comment]
You can fork it and build a custom Package by yourself

**Comment by crdmrn** [comment]
@turadr Yes please, the repo at this point is practically unusable for production, as it comes with an enormous amount of unneeded stuff.
A package integration would (and should) the way to go.

@pedro15 what kind of a jerk answer is that?
This package is even mentioned on the Unity official documentation, and somehow still it's completely out of the intended usage for extensions that Unity itself is trying to push to the users: packages.
Of course everyone could do their own fork and make a package out of it but, given the situation, it's on them to make it available through the package manager as they did with the PostProcessing stack, for example.

Otherwise just mark the whole thing as deprecated.

**Comment by pedro15** [comment]
> 
> 
> @turadr Yes please, the repo at this point is practically unusable for production, as it comes with an enormous amount of unneeded stuff (e.g. all the base project folders; if I only want the base code I have to delete half of the repo).
> A package integration would be (and should be) the way to go.
> 
> @pedro15 what kind of a jerk answer is that?
> This package is even mentioned on the Unity official documentation, and somehow still it's completely out of the intended usage for extensions that Unity itself is trying to push to the users: packages.
> Of course everyone could do their own fork and make a package out of it but, given the situation, it's on them to make it available through the package manager as they did with the PostProcessing stack, for example.
> 
> Otherwise just mark the whole thing as deprecated.

Yes, but we know unity is begin unity. They don't listen to the community very much as i know. 
Take the example of how many years it took to them to make their shader editor. or give de dark skin for free users?
I don't even complain very much in this sort of things. at least we can do alternatives while we don't know when this will be avaiable for us.

**Comment by turadr** [comment]
We are currently working on delivering these components as a package.

**Comment by marwie** [comment]
Could we leave this issue open to track progress?

**Comment by turadr** [comment]
Alright.

**Comment by crdmrn** [comment]
> We are currently working on delivering these components as a package.

@turadr is there a rough estimate on when that might happen?

**Comment by turadr** [comment]
The package should be ready most probably in the next few weeks.

**Comment by xCyborg** [comment]
@turadr I hope it finally will, after more than 3 years of waiting.

**Comment by turadr** [comment]
I'm happy to let you know that in the branch [package](https://github.com/Unity-Technologies/NavMeshComponents/tree/package) you'll find the code ready to be added in your projects through the Package Manager. Please follow the instructions in the [Setup](https://github.com/Unity-Technologies/NavMeshComponents/tree/package#setup) section of the README.

Please let us know if you encounter any issues when using the package. 

Next we'll be working on publishing the package in the official registry in the beginning of 2021.

**Comment by marwie** [comment]
@turadr awesome, thanks for the effort and update

**Comment by turadr** [comment]
We've published the first experimental version of the NavMesh Components package to the Unity Registry. You can use it in your projects by adding the following dependency in your `Packages\manifest.json` files:
`    "com.unity.ai.navigation.components": "1.0.0-exp.1"`

**Comment by crdmrn** [comment]
Nice! @turadr are there plans to make it available in the official Unity registry, among the other packages (so that there's no need to manually add it to the manifest.json?)

**Comment by turadr** [comment]
It is already available in the Unity Registry only it has been hidden for the time being. I don't know when it will be added to the visible list because we're considering making a few improvements to it first. Most probably not this year (not for 2021.2).
Until then you can add it from the UI by filling in the package name in the `Add package from Git URL...` (2020.2+) or `Add package by name...` (2021.1+) Package Manager menus.

![image](https://user-images.githubusercontent.com/9038468/112617902-d85f7200-8e25-11eb-8065-cca66e0fcf0d.png)
![image](https://user-images.githubusercontent.com/9038468/112618093-165c9600-8e26-11eb-80b8-38fd3f9294bc.png)
![image](https://user-images.githubusercontent.com/9038468/112618677-ae5a7f80-8e26-11eb-9951-10ff72061987.png)

**Comment by turadr** [comment]
The package has been renamed to  _AI Navigation_ - `com.unity.ai.navigation` 1.0.0-exp.1 . 
If I'm not mistaken, using the old name still works but the Package Manager will load up this new package.
